### ISSUE TITLE
Can't import to Android Studio

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [edsilfer/sticky-index#1]
It says :
"The project 'stickey-index' is not a Gradle-based project."

Please fix this.

Also, btw, you can put on gradle/maven by using this: https://jitpack.io/ . Someone wrote me about this from my library: https://github.com/AndroidDeveloperLB/MaterialPreferenceLibrary/issues/4

**Comment by edsilfer** [comment]
I'm having a hard time trying to upload it to JCenter/Maven but I am working on it. I tried to use https://jitpack.io/ but I keep gettin this erro:Error:(24, 13) Failed to resolve: com.github.edsilfer:sticky-index:05ed29c08f

**Comment by AndroidDeveloperLB** [comment]
So even there it's hard? I never tried it, but it looks quite easy to do it there...

**Comment by edsilfer** [comment]
Yes, I agree! I can even find my repo in their page, however even importing their private maven repo and using the link they provided for gradle, I got the error that I mentioned above. It's been three days that I try to upload it in JCenter with no success

**Comment by jitpack-io** [comment]
@edsilfer Submitted PR #2 to fix. Gradle build needed some small tweaks that we describe [here](https://github.com/jitpack/jitpack.io/blob/master/ANDROID.md).

**Comment by AndroidDeveloperLB** [comment]
wait, so it works?

**Comment by jitpack-io** [comment]
It does now:) https://jitpack.io/#edsilfer/sticky-index/a723b997ad

**Comment by edsilfer** [comment]
Yes, it is working now! I've just finishing testing here. Thank you @jitpack-io , I will certainly advice the use of you guys solution, it saved me a lot of time :D. A last question, how can I create a version, so it wont pick that ugly commit code?

Thank you for letting me know about jitpack @AndroidDeveloperLB

**Comment by jitpack-io** [comment]
Great to hear that! For versions, just create a GitHub release https://github.com/edsilfer/sticky-index/releases That will also create a git tag in your repo.

**Comment by edsilfer** [comment]
Awesome, release 1.0.0 beta online. Thank you again @jitpack-io :D

**Comment by AndroidDeveloperLB** [comment]
It doesn't work for me. It shows this :
![untitled20150607003512](https://cloud.githubusercontent.com/assets/5357526/8021565/1958c208-0cad-11e5-96bb-a76bef39f106.png)

What should I do?

**Comment by jitpack-io** [comment]
You should just add the dependency to your build.gradle as shown here: https://jitpack.io/#edsilfer/sticky-index/1.0.0

**Comment by AndroidDeveloperLB** [comment]
@jitpack-io Are you writing this to me to to @edsilfer ?
I just used the github link. It doesn't allow to import it.

**Comment by edsilfer** [comment]
@AndroidDeveloperLB , I've just update the read-me, see if the **how to** section helps you. 

1. You first create a new project on Android Studio (or use an existing one)
2. Go to the build.gradle and add: 

```
repositories {
    maven {
        url "https://jitpack.io"
    }
}

dependencies {
    compile 'com.github.edsilfer:sticky-index:1.0.0'
}
```

3. Rebuild your project;
4. Follow the **how to** section from read-me

**Comment by AndroidDeveloperLB** [comment]
@edsilfer I just want to try out the sample&library you've made. All libraries allow to do it by just using the git URL.
Yours is :
https://github.com/edsilfer/sticky-index.git
so what I do is to go to Android-Studio, choose VCS->Checkout from version control-> Git , and then enter this URL.

**Comment by edsilfer** [comment]
@AndroidDeveloperLB  now I get it. I am currently using gradle v2.4 and it seems that some major changes happened with this version that cause conflicts with some other stuff. I'm currently searching a solution for this and once I figure it out I will let you know

**Comment by edsilfer** [comment]
@AndroidDeveloperLB , problem solved. When checking in the project, you got pick the gradle home folder - if possible, gradle version 2.4 (I know that before gradle 2.0 JCenter was not even included in the libs). In case you don't have it installed you can use:

```
$ sudo add-apt-repository ppa:cwchien/gradle
$ sudo apt-get update
$ sudo apt-get install gradle
```
After pointing to gradle home directory, you can proceed with the wizard (I've just tested out it here and work well). Let me know if you still have problems

**Comment by AndroidDeveloperLB** [comment]
@edsilfer I just use the Gradle plugin of Android Studio. They advance with it in the pace that matches their own advancements, and they also warn about using too new versions of it.
I've ignored the dialog I've shown, and re-opened the project. Now it allows me to run it. You should fix the project so that everyone will be able to import it.
Now it works fine. Reminds me of my repo here (for listView): 
https://github.com/AndroidDeveloperLB/ListViewVariants
and for fast-scrolling on recyclerView, I made this one:
https://github.com/AndroidDeveloperLB/LollipopContactsRecyclerViewFastScroller

**Comment by edsilfer** [comment]
@AndroidDeveloperLB I had some issues with the gradle plugin that came with AS. I've read in some question - probably in stackoverflow - that the version I was using was a really old version of graddle. I remember I was getting problems running the build 'cause the version that came with my AS didn't include the JCenter refferences or smtg like that. 

Yes, looking at your libraries it does look similar. I've searched a lot but the only reference to thiss subject that I've found were here: http://stackoverflow.com/questions/27621425/how-to-mimic-the-listview-stickey-items-like-on-lollipops-contacts-app and here http://stackoverflow.com/questions/26957313/how-to-make-a-listview-like-this-image. As I could not find any library that fits the RecyclerView, I decided to make my own. 

The issues with the transion has to do with fast scrolling righ? Sometimes the sticky index vanishes if you scroll too fast. Well I'll try to fix it when I have some time, I will also add the layout orientation in the know issues. Thanks for your feedback.
